### ISSUE TITLE
allow archiving / retiring pipelines 

### ISSUE THREAD

**Issue by tkellen** [concourse/concourse#2518]
# Feature Request
I'd like the ability to archive a pipeline without deleting it so the build history / links are still usable but the UI isn't cluttered.

## A Modest Proposal
This seems like a pretty easy series of pull requests. Here's what I have in mind:

1. Create a migration adding an `archived` boolean field to the pipelines table.
2. Add support to the pipelines endpoint in atc to filter on archive status for GET and modify archive status with PUT.
3. Add `archive-pipeline` and `unarchive-pipeline` commands to fly (basically a cut/paste job of hide/expose)
4. Add a toggle in the UI (exactly the same as the high density toggle) to control if archived pipelines or active pipelines are currently being viewed.

What do ya'll think?

PS: Thanks so much for making concourse, I am really enjoying using it!

**Comment by tkellen** [comment]
The PRs associated with this issue now provide minimal API and CLI support for this functionality. 

No harm no foul if this turns out to be an undesirable feature. This was an interesting exploration of the concourse source code for me.

Assuming this is something ya'll would like to see through, more work needs to be done on test coverage, support needs to be added to the UI, and I'm sure there are plenty of implications I haven't considered that should be discussed.

Standing by!

**Comment by vito** [comment]
Neat, I like the idea. One thing to consider is that we should *probably* have archived pipelines become effectively paused, right?

Glad to see it wasn't too hard to dive in to the code! FWIW, we're planning to do a [big restructuring](https://discuss.concourse-ci.org/t/psa-the-great-code-restructing-of-2018/543/3) that'll hopefully make it much easier.

**Comment by tkellen** [comment]
Oh good call. Yeah, pausing would definitely be warranted. Would unpausing be disabled until a pipeline is unarchived, or would unpausing implictly unarchive a pipeline?

**Comment by tkellen** [comment]
Also, I just read your note about restructuring. Kudos! The unix philosophy is fantastic. Applying it to git repositories is a nightmare for most, though. You can use folders to accomplish the same thing people :rofl:! Monorepo for the win.

**Comment by tkellen** [comment]
@vito Got it. We'd also probably want this as a derived field in the API responses so we can have the UI similarly check for "active" when displaying paused state, right?

**Comment by tkellen** [comment]
Archived status having a wholly separate (and superseding) representation works for me and keeps the implementation a bit more straightforward. I think the toggle still makes sense, but I am happy to hold off on any UI work until @Lindsayauchin weighs in.

I can proceed with further updates to support archive mode implicitly pausing. Can you tell me precisely which layer of abstraction you'd like .`Active` to be represented?

**Comment by tkellen** [comment]
Any thoughts @Lindsayauchin? Thanks in advance!

**Comment by Lindsayauchin** [comment]
@tkellen - Awesome idea! Great that you are working on this!

Couple questions for you on your use case: 
 - Can you give an example of the last time you wanted to do this and what did you do to work around this?
- How often are would you use this feature?
- Are you expecting a view of all the archived pipelines in your team or across the instance?
- How deep are you expecting to go in the read-only pipeline? ie. how far in the history of builds?
- Are you expecting to unarchive a pipeline from the UI?

**Comment by tkellen** [comment]
Hi @Lindsayauchin. Thanks for the reply :dancer:, I'll jump right in:

1. Example of last time I wanted to do this: when I was ready to discard a pipeline that is no longer needed technically but necessary to retain the history of for audit purposes. The workaround was renaming it. My actual use case is watching a repo and creating a new pipeline for each branch and archiving it when the branch is deleted (maybe someday https://github.com/concourse/concourse/issues/1707 will obviate this need but it looks to be very far off).
2. Daily.
3. By team.
4. Access to the full history seems warranted?
5. No.

**Comment by Lindsayauchin** [comment]
@tkellen @vito 

With the introduction of the tags for meber or exposed teams we could use the area beside this element to show if there are archieved pipelines for that team with a counter. 

A user could then select this to display the archived pipelines with the current active pipelines. 

![screen shot 2018-09-20 at 12 06 59 pm](https://user-images.githubusercontent.com/27746614/45831566-b83b1f00-bccd-11e8-812d-129574a6ad5a.png)

![screen shot 2018-09-20 at 12 06 46 pm](https://user-images.githubusercontent.com/27746614/45831570-bb360f80-bccd-11e8-9d89-bc0bdbfa0651.png)


The pipeline cards on the dashboard view would have some new UI elements such as: an icon for archived displayed where the timestamp normally appears, the display of the thumbnail dimmed down to indicated it is no longer active, the running animation stripe displaying grey to represent inactive.  

<img width="300" alt="screen shot 2018-09-20 at 11 44 52 am" src="https://user-images.githubusercontent.com/27746614/45830202-a1df9400-bcca-11e8-93ee-1a82c9813072.png">

As the user drills down to the buildpage from the dashboard the breadcrumbs will display the archived icon preceeding the pipeline name. 

![pipeline_archived](https://user-images.githubusercontent.com/27746614/45830095-62b14300-bcca-11e8-84be-aec6f9b697b5.png)

User will have a read-only view of the build page and the trigger button will not be displayed. 

![pipeline_build_page_archived](https://user-images.githubusercontent.com/27746614/45830099-6644ca00-bcca-11e8-8603-e85ff5e523ff.png)

**Comment by vito** [comment]
Looks great to me!

**Comment by tkellen** [comment]
Nice! I think I can run with that. I tried bringing up the new monorepo to re-open my PR and ran into:
https://github.com/maxbrunsfeld/counterfeiter/pull/94

@vito, will there be an announcement of some kind when you're ready for contributors again? I am following #2534.

**Comment by vito** [comment]
@tkellen Yeah, I'll post an announcement somewhere when things are all ready. Probably the #contributors channel on Discord, and maybe a few other places. I can help you along if you want to keep working on it soon. I'll update the [contributing docs](https://github.com/concourse/concourse/blob/master/.github/CONTRIBUTING.md) once things are smoother.
